core/src/replay_stage.rs

```diff
@@ -3559,6 +3566,7 @@ impl ReplayStage {
                 && propagation_confirmed
                 && switch_fork_decision.can_vote()
             {
+                assert_eq!(check_slot, bank.slot());
```

Comment:
Not sure a logging slot wrong mismatching is worth a panic?

Reply:
i think it should panic because if we vote on a slot other than `check_slot` then we might be committing a slashable offense.